File: gen-sig-pkcs1_padding.py
#!/usr/bin/python3

import base64
import sys
import time

from Crypto.PublicKey import RSA

from Crypto.Random import get_random_bytes
from Crypto.Cipher import PKCS1_v1_5
#from Crypto.Cipher import PKCS1_OAEP

mytime = str(int(time.time()))

key = RSA.import_key(open('testing-public.key').read())

cipher = PKCS1_v1_5.new(key)
#cipher = PKCS1_OAEP.new(key)

sig1 = cipher.encrypt( mytime.encode() );

sig2 = base64.b64encode( sig1 ).decode()

print("Timestamp being encrypted: {ts}\nThe signature is below:\n{sig}".format(ts=mytime,sig=sig2))

File: gen-sig-pkcs1_padding.pl
#!/usr/bin/perl -w

use strict;
use warnings;

use File::Slurp;
use Crypt::OpenSSL::Random;
use Crypt::OpenSSL::RSA;
use MIME::Base64;

my $key_string = read_file('testing-public.key');

my $plaintext = time();

my $rsa_pub = Crypt::OpenSSL::RSA->new_public_key($key_string);

# use_pkcs1_padding
# use_pkcs1_oaep_padding
# use_sslv23_padding
# use_sha256_hash
# use_sha512_hash
# use_sha384_hash
#

#$rsa_pub->use_sslv23_padding(); # use_pkcs1_oaep_padding is the default

$rsa_pub->use_pkcs1_padding();

my $ciphertext = encode_base64($rsa_pub->encrypt($plaintext),'');

print "Timestamp being encrypted: $plaintext\nThe signature is below:\n";
print "$ciphertext \n\n";
exit
